src/worker.cc

```diff
@@ -175,8 +175,10 @@ rapidjson::Document from_string(const std::string& json, const valhalla_exceptio
 
 void add_date_to_locations(Options& options,
                            google::protobuf::RepeatedPtrField<valhalla::Location>& locations) {
-  // otherwise we do what the person was asking for
-  if (options.has_date_time_case() && !locations.empty()) {
+  // /(trace_)route needs special treatment
+  if (options.has_date_time_case() &&
+      (options.action() == Options::route || options.action() == Options::trace_route) &&
```

Comment:
I wasn't aware that `trace_route` cares about `date_time.type` settings.. though I see it doesn't do anything with type = 2 (arrive_by), which wouldn't make much sense. `trace_attributes` probably shouldn't get the global `date_time` string either if it's set, that hardly makes sense I think?

Reply:
> which wouldn't make much sense
are you trying to say that it seems correct to you that it doesnt handle `arrive_by`? or are you saying it doesnt seem right to you that it doesnt handle it?

why are we changing this logic anyway? what was wrong with just setting it regardless of the action? sure some actions dont support it but they are blocked at other parts in the request processing

Reply:
I guess that's answered by your comment below. Essentially `date_time.type` really only has a meaning when a route is requested with an origin & destination. Then we want to force set a time on first/last location or it's invariant (for bidir astar).

Reply:
> are you trying to say that it seems correct to you that it doesnt handle arrive_by?

right, this one:) sorry, was a bit unclear, I see that now..